web3/eth.py

```diff
@@ -85,68 +89,88 @@ def coinbase(self):
     def mining(self):
         return self.web3.manager.request_blocking("eth_mining", [])
 
+    @property
+    def hash_rate(self):
+        return self.web3.manager.request_blocking("eth_hash_rate", [])
+
     @property
     def hashrate(self):
-        return self.web3.manager.request_blocking("eth_hashrate", [])
+        raise DeprecationWarning("This method has been deprecated")
+
+    @property
+    def gas_price(self):
+        return self.web3.manager.request_blocking("eth_gas_price", [])
 
     @property
     def gasPrice(self):
-        return self.web3.manager.request_blocking("eth_gasPrice", [])
+        raise DeprecationWarning("This method has been deprecated")
 
     @property
     def accounts(self):
         return self.web3.manager.request_blocking("eth_accounts", [])
 
     @property
-    def blockNumber(self):
-        return self.web3.manager.request_blocking("eth_blockNumber", [])
+    def block_number(self):
+        return self.web3.manager.request_blocking("eth_block_number", [])
 
     @property
     def chainId(self):
         return self.web3.manager.request_blocking("eth_chainId", [])
 
-    def getBalance(self, account, block_identifier=None):
+    def get_balance(self, account, block_identifier=None):
```

Comment:
We'll want to follow the same pattern that is in place in the Geth module. So we'll need to add a utils file (probably called `web3/_utils/eth.py`, and define variables like `get_balance = Method('eth_getBalance', mungers=[...])`. Then we'll also need to add the deprecated methods like: `getBalance = DeprecatedMethod(...)`. Then we'll need to import those into this file like we do [here](https://github.com/ethereum/web3.py/blob/master/web3/geth.py#L5). We'll also need to make a class like we do [here](https://github.com/ethereum/web3.py/blob/master/web3/geth.py#L80). Hopefully that's enough to get you going, but let me know if anything is unclear!